src/utils/tokenCounter.ts

```diff
@@ -0,0 +1,125 @@
+import { get_encoding, Tiktoken } from "tiktoken"
+
+/**
+ * Shared token counter utility using tiktoken's cl100k_base encoding
+ * This provides a universal proxy token counter that works reasonably well for most models
+ */
+class TokenCounter {
+	private encoding: Tiktoken | null = null
+	private initializationPromise: Promise<void> | null = null
+
+	/**
+	 * Initialize the tiktoken encoding (lazy loading)
+	 */
+	private async initialize(): Promise<void> {
```

Comment:
Why does it call a promise version of this? Can't we just create a new Tiktoken(o200k_base) instance directly in a constructor when this new class is created by the caller?